### PR TITLE
perf: incoming tab take some seconds to load data

### LINKED ISSUE #1854: [Feature] When the user click on Incoming tab take some seconds to load data. Should show instantly.
### Feature description

For now, when the user click on Incoming tab take some seconds to load the transactions. I think Alby could cache the data of incoming transactions, so show instantly to the user when clicking on incoming tab. And a loading icon to the user know that Alby is still looking for new transactions.

### Describe the solution

I think Alby could cache the data of incoming transactions, so show instantly to the user when clicking on incoming tab. And a loading icon to the user know that Alby is still looking for new transactions.

### Describe alternatives

_No response_

### Additional context

_No response_

### Are you working on this?

No

### PR DESCRIPTION
**Pull request by qqqzhch**
### Describe the changes you have made in this PR

Load incoming txs as soon as popup is being opened 

if account is being switched Load incoming

### Link this PR to an issue [optional]

- Fixes #1854

### Type of change

(Remove other not matching type)

- `fix`: Bug fix (non-breaking change which fixes an issue)

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

**Comment by escapedcat** [comment]
This works, data is there immediately. But it might not be the "latest" data, right?
I feel like step 2 is missing in a way: https://github.com/getAlby/lightning-browser-extension/issues/1854#issuecomment-1345523880

**Comment by qqqzhch** [comment]
> This works, data is there immediately. But it might not be the "latest" data, right? I feel like step 2 is missing in a way: [#1854 (comment)](https://github.com/getAlby/lightning-browser-extension/issues/1854#issuecomment-1345523880)

If the pop page remains open for a long time, the data is not updated.
If the balance changes and the data is loaded again it should be much better, I'll try it

**Comment by qqqzhch** [comment]
The relevant test code for this component is still missing and I will add the relevant test code after this pr merge.
Can the e2e tests be run only if the local Lightning Network node is linked, or the data be mocked and run without linking the node?
metamask has also made a test site for testing dapp related functions, I think we should also make a similar site
https://metamask.github.io/test-dapp/

**Comment by escapedcat** [comment]
You can have a look at [our e2e tests](https://github.com/getAlby/lightning-browser-extension/tree/master/tests/e2e), we use testnet there.  
I.e. here: https://github.com/getAlby/lightning-browser-extension/blob/master/tests/e2e/002-walletFeatures.spec.ts/#L88-L109

**Comment by escapedcat** [comment]
> If the balance changes and the data is loaded again it should be much better

Ah, interesting. Did you test this? Does it work?

File: src/app/screens/Home/DefaultView/index.tsx [hunk 1]
```diff
@@ -161,7 +156,14 @@ const DefaultView: FC<Props> = (props) => {
 
     setIncomingTransactions(invoices);
     setIsLoadingInvoices(false);
-  };
+  }, [getFormattedFiat, settings]);
```
Comment (im-adithya) [inline src/app/screens/Home/DefaultView/index.tsx:159]: `settings.showFiat` would suffice right?

**Review by im-adithya — APPROVED** [review]
Nice improvement @qqqzhch! Works as expected, even on balance change!
(Left a small change)

**Comment by qqqzhch** [comment]
@escapedcat @im-adithya 
I have not done enough tests on pages that are opened and stay particularly long, not yet
- 1 Now only repeated transactions between multiple accounts in one wallet, did the test
- 2 I think I still have to install two wallets in two browsers, and keep the wallets inside both browsers open for a long time, to do tests of transactions

File: src/app/screens/Home/DefaultView/index.tsx (hunk 1 above)
Comment (qqqzhch) [reply src/app/screens/Home/DefaultView/index.tsx:159]: Good suggestion settings.showFiat is better

**Comment by im-adithya** [comment]
I went with the assumption that if we're not updating the balance real-time, there's no point in showing the recent transactions (which didn't even reflect in the balance, as that might lead to confusion).

So I think as long as we have balance (`balancesDecorated?.accountBalance`) as a dependency, we're fine!

@escapedcat thoughts?

**Comment by escapedcat** [comment]
Tested with sending some sats from an account to another account. The balance is only updated on "reopening" the extension anyways. So unless we do not have any "polling" I guess adding the effect-stuff is useless? In the case of "an invoice is being payed" this works though. So I guess it's still good to have it.

**Comment by qqqzhch** [comment]
Yes, come to think of it, the balance does not change instantly issue, there are too many types of wallet connections, for different wallet connections my knowledge is relatively low, to read in real time, need all connect  support wss, or use setInterval

**Review by escapedcat — APPROVED** [review]

**Comment by escapedcat** [comment]
> Yes, come to think of it, the balance does not change instantly issue, ...

I think we're good here for now.

**Pull request merged**